.github/workflows/auto-update-gemfiles.yml

```diff
@@ -0,0 +1,139 @@
+name: Auto-update Gemfiles
+
+# This action cannot be skipped altogether because it is a mandatory status check.
+# Instead we conditionally skip it at job level, instead of workflow level.
+on:
+  push:
+    branches:
+      - quinna/update-gemfiles
+  workflow_dispatch:
+    
+# Ensure obsolete job is cancelled if another commit is pushed to the same branch.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    permissions:
+        actions: read # read secrets
+        pull-requests: write
+        contents: write
+    strategy:
+        fail-fast: false
+        matrix:
+            engine:
+              # ADD NEW RUBIES HERE
+              - name: ruby
+                version: '3.4'
+              - name: ruby
+                version: '3.3'
+              - name: ruby
+                version: '3.2'
+              - name: ruby
+                version: '3.1'
+              - name: ruby
+                version: '3.0'
+              - name: ruby
+                version: '2.7'
+              - name: ruby
+                version: '2.6'
+              - name: ruby
+                version: '2.5'
+              # - name: jruby
+              #   version: '9.4'
+              # - name: jruby
+              #   version: '9.3'
+              # - name: jruby
+              #   version: '9.2'
+    container:
+      image: "ghcr.io/datadog/images-rb/engines/${{ matrix.engine.name }}:${{ matrix.engine.version }}"
+    steps:
+      # Only execute if there's a PR attached to this branch.
+      # Because we execute on `push`, we have to double check here if this is part of a PR.
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Fetch all history to access all branches
+
+      - name: Output Ruby version
+        run: |
+          echo "Ruby version: $RUBY_VERSION"
+      
+      - name: Build gem
+        run: |
+            bundle install
+            bundle exec rake edge:update
+      
+      - name: Store modified gemfiles
+        id: mod_files
+        run: |
+          # Example command to create a tar.gz file of modified files
+          mkdir -p modified_gemfiles
+          VERSION=${{ matrix.engine.version }}
+          find . -name "*${VERSION}_rack_*.gemfile" -exec cp --parents {} modified_gemfiles/ \;
```

Comment:
copy files in modified_gemfiles/, cp --parents is equivalent of mkdir -p preserves directory structure